src/quo2/components/graph/wallet_graph/view.cljs

```diff
@@ -0,0 +1,70 @@
+(ns quo2.components.graph.wallet-graph.view
+  (:require [quo2.theme :as theme]
+            [react-native.linear-gradient :as linear-gradient]
+            [react-native.charts :as charts]
+            [react-native.core :as rn]
+            [quo2.components.graph.wallet-graph.style :as style]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.graph.wallet-graph.utils :as utils]))
+
+(defn- max-data-points
+  [time-frame]
+  (case time-frame
+    :empty    0
+    :1-week   7
+    :1-month  30
+    :3-months 90
+    :1-year   365
+    500))
+
+(defn- view-internal
+  [{:keys [data state time-frame theme]}]
+  (let [max-data-points (max-data-points time-frame)
+        data            (if (and (not= time-frame :empty) (> (count data) max-data-points))
+                          (utils/downsample-data data max-data-points)
+                          data)
+        max-value       (when-not (= time-frame :empty) (utils/find-highest-value data))
+        width           (:width (rn/get-window))
+        line-color      (if (= state :positive)
+                          (colors/theme-colors colors/success-50 colors/success-60 theme)
+                          (colors/theme-colors colors/danger-50 colors/danger-60 theme))
+        gradient-colors [(colors/alpha line-color 0.1) (colors/alpha line-color 0)]
+        fill-color      (colors/theme-colors colors/white colors/neutral-95)]
+    [linear-gradient/linear-gradient
+     {:colors gradient-colors
+      :start  {:x 0 :y 1}
+      :end    {:x 0 :y 0}
+      :style  style/gradient-background}
+     (if (= time-frame :empty)
+       [rn/view
+        {:accessibility-label :illustration
+         :style               style/illustration}
+        [text/text "Illustration here"]]
+       [rn/view {:accessibility-label :line-chart}
+        [charts/line-chart
+         {:height                  96
+          :width                   (+ width 1)
+          :max-value               max-value
+          :min-value               0
+          :adjust-to-width         true
+          :data                    data
+          :area-chart              true
+          :start-fill-color        fill-color
+          :end-fill-color          fill-color
+          :hide-data-points        true
+          :hide-rules              true
+          :hide-y-axis-text        true
+          :x-axis-indices-height   100
+          :thickness               2
+          :color                   line-color
+          :y-axis-thickness        0
+          :x-axis-thickness        0
+          :initial-spacing         0
+          :end-spacing             0
+          :disable-scroll          true
+          :y-axis-label-width      0.01
+          :labels-extra-height     -36
+          :x-axis-label-text-style style/x-axis-label-text-style}]])]))
+
+(def view (theme/with-theme view-internal))
```

Comment:
nit: let's use `quo.theme` as `theme` can clash here and is technically and unsafe usage. i.e `[quo2.theme :as quo.theme]`

Reply:
Done